### ISSUE TITLE
Reading: Understanding Measures of Uncertainty for Adversarial Example Detection

### ISSUE THREAD

**Issue by ayaco0** [ayaco0/paper-survey#38]
### 論文リンク
https://arxiv.org/abs/1803.08533

### 著者/所属機関
Lewis Smith, Yarin Gal

### 投稿年
2018

## 研究背景
- 敵対的事例とは
  - 意図的に誤った予測や過信を招く入力データのこと
  - 正常?な学習データの多様体から距離が離れたところに存在する傾向がある
- 多様体への距離を測定する手法
  - 最近傍法: ピクセル単位の距離は知覚的な類似性の指標として不十分❌
  - 密度モデル: 画像認識で使われる高次元空間に拡張するのが難しい❌
- 「多様体から離れた事例は不確実性が大きい」＝ 不確実性を用いて敵対的事例の検出が可能？🤔

<img width="354" alt="スクリーンショット 2024-01-15 23 25 54" src="https://github.com/ayaco0/paper-survey/assets/74520178/0ab81e3a-4d49-46ef-92c3-90ff2aa5ae06"> 

- 不確実性の推定方法: MC dropout 
  - ドロップアウトを用いてベイズ推定を近似する
- 不確実性を測る指標: 予測エントロピー，相互情報量，softmaxの分散，期待?エントロピーなど

## ベイジアンNN
- ディープNNの重みは(1)式みたいな感じ（weight decay = λ の L2正則化）

![スクリーンショット 2024-01-16 8 18 07](https://github.com/ayaco0/paper-survey/assets/74520178/ff4f7efa-ad36-4809-9fd6-7164580ed4fc)

- 通常のNNの重みは最適化された固定パラメータ→点推定?
- ベイジアンNNの重みは確率変数として扱う
  - 重みに事前分布p(ω) を与える
  - 点推定じゃなくて事後分布を推定するので不確実性を得られる

## 変分推論とドロップアウト
- 近似することで積分計算しない単純な最適化問題に変換できる?
- ドロップアウトではNNの一部のノードをランダムにゼロにする
- 出力をサンプリングしていけば期待値を近似できる

![スクリーンショット 2024-01-16 8 39 54](https://github.com/ayaco0/paper-survey/assets/74520178/2de415dd-f5d8-4ded-884d-8d606cb9099b)

## 不確実性の指標
- モデルの不確実性は2種類ある
  - 1. 認識論的不確実性（epistemic uncertainty）：知識不足による不確実性
      - データ不足が原因でモデルのパラメータが決まらない＝パラメータの事後分布も広くなる
  - 2. 偶然的?不確実性　（aleatoric uncertainty）：データの真の確率による不確実性
      - 学習データ増やしても解決せず，予測そのものが難しいデータによる不確実性
- epistemic と aleatoric を区別できたら再学習すればいいのかデータをクリーニングすればいいのかわかる？
- 予測エントロピー
  - これだけだとepistemic と aleatoric を区別できない
<img width="429" alt="スクリーンショット 2024-01-16 9 07 02" src="https://github.com/ayaco0/paper-survey/assets/74520178/36dc6c06-bfcb-4995-a5c9-fd04f54577ac">

- 相互情報量
  - epistemic のほうを定量化できる指標
<img width="393" alt="スクリーンショット 2024-01-16 9 08 16" src="https://github.com/ayaco0/paper-survey/assets/74520178/943120f8-7d2d-42ba-be9c-bb25fe9ff7fc">
<img width="392" alt="スクリーンショット 2024-01-16 9 25 11" src="https://github.com/ayaco0/paper-survey/assets/74520178/daabe09d-00cd-4ae8-9668-45e575dca3ea">

- 相互情報量とソフトマックス分散はほぼ等しい
<img width="341" alt="スクリーンショット 2024-01-16 9 56 39" src="https://github.com/ayaco0/paper-survey/assets/74520178/b1921092-93c1-4aaf-a7e9-758f28dedc2a"> <img width="355" alt="スクリーンショット 2024-01-16 9 56 19" src="https://github.com/ayaco0/paper-survey/assets/74520178/1ccd21f9-978f-403e-b75b-09e262d31567">

## 敵対事例検出における不確実性指標の有効性検証実験
- 予測エントロピーと相互情報量の挙動の違いとかも見ていく
- 以下のデータセットで検証
  - MNIST（手書き文字認識）の評価データセット
  - Kaggleの犬猫識別データセット

### 結果
<img width="353" alt="スクリーンショット 2024-01-15 23 26 30" src="https://github.com/ayaco0/paper-survey/assets/74520178/999500b0-a408-4a50-9df2-9e2e8510161b"> 
<img width="358" alt="スクリーンショット 2024-01-15 23 26 18" src="https://github.com/ayaco0/paper-survey/assets/74520178/ba63e96b-7dda-41ea-8479-81c9766a51dd"> 

- （左）相互情報量は多様体にある画像と外れてる画像の区別ができてる（らしい）
  - どう見ればいいの？相互情報量の方が画像空間でのピークが分かりやすくはある
  - 予測エントロピーだとできてない
- （右）相互情報量は潜在空間と画像空間で異なる統計的特性が現れる
  - 画像空間ならではの特徴をよく捉えられるみたいなことか？

<img width="300" alt="スクリーンショット 2024-01-16 14 04 28" src="https://github.com/ayaco0/paper-survey/assets/74520178/f01ffbc7-07ee-4256-9eb9-84cd5c72de68"> <img width="300" alt="スクリーンショット 2024-01-16 14 04 44" src="https://github.com/ayaco0/paper-survey/assets/74520178/27961698-0ba2-4729-8692-8509f20716fa">

- （左）予測エントロピーによる不確実性，（右）相互情報量による不確実性
  -  黒いほど不確実性が小さい
  - 相互情報量はクラスタの境界で不確実性が大きくなっている一方，エントロピーは小さく
  - 学習データから離れてるにもかかわらずエントロピーが大きくて相互情報量が小さい領域がある

<img width="354" alt="スクリーンショット 2024-01-15 23 27 00" src="https://github.com/ayaco0/paper-survey/assets/74520178/16191dc7-31e7-4d3f-9f52-2036ca8f4691">  <img width="356" alt="スクリーンショット 2024-01-15 23 27 25" src="https://github.com/ayaco0/paper-survey/assets/74520178/e2ead313-fbe9-4d71-a254-abecb3713d55">  

- 
<img width="725" alt="スクリーンショット 2024-01-15 23 27 49" src="https://github.com/ayaco0/paper-survey/assets/74520178/e9e1e759-cb4e-411c-b64f-a3b539dc4195">

- 

<img width="722" alt="スクリーンショット 2024-01-15 23 28 04" src="https://github.com/ayaco0/paper-survey/assets/74520178/ba2fa6a2-5942-413b-a0b7-c96ccab3c87b">

- 

## 感想
- 密度モデルってなあに
- 予測エントロピーがダメで相互情報量が良い理論的な理由がまだ理解できない

## 参考
- [[DL輪読会]Understanding Measures of Uncertainty for Adversarial Example Detection](https://www.slideshare.net/DeepLearningJP2016/dlunderstanding-measures-of-uncertainty-for-adversarial-example-detection)
